lib/rubygems_plugin.rb

```diff
@@ -0,0 +1,277 @@
+# frozen_string_literal: true
```

Comment:
There are exceptions to use `rubygems_plugins`, 
but almost changes of this file is what it is copied from `lib/rdoc/rubygems_hook.rb`.

<details>
  <summary>diff</summary>
  
  ```diff
  ❯ git diff ef92423ef1b6492bcec7d94971d63b87f5adf6c5:lib/rubygems_plugin.rb upstream/master:lib/rdoc/rubygems_hook.rb
  diff --git a/lib/rubygems_plugin.rb b/lib/rdoc/rubygems_hook.rb
  index e2a4af76..3160072e 100644
  --- a/lib/rubygems_plugin.rb
  +++ b/lib/rdoc/rubygems_hook.rb
  @@ -1,28 +1,15 @@
   # frozen_string_literal: true
  -
   require 'rubygems/user_interaction'
   require 'fileutils'
  -
  -require_relative 'rdoc'
  +require_relative '../rdoc'
  
   ##
   # Gem::RDoc provides methods to generate RDoc and ri data for installed gems
   # upon gem installation.
   #
   # This file is automatically required by RubyGems 1.9 and newer.
  -#
  -# Difference between RubygemsHook and RubyGemsHook.
  -#   - RubygemsHook is executed from RubyGems.
  -#   - RubyGemsHook is executed from rubygems_plugin.
  -#
  -# The reason why we use two classes.
  -# If there are classes named 'RubygemsHook',
  -# conflicts of name may happen.
  -#
  -# We can add methods to existing RubygemsHook,
  -# but we didn't adopt it because of less mentenability and create new class 'RubyGemsHook'.
  -#
  -class RDoc::RubyGemsHook
  +
  +class RDoc::RubygemsHook
  
     include Gem::UserInteraction
     extend  Gem::UserInteraction
  @@ -58,7 +45,7 @@ class RDoc::RubyGemsHook
     # Post installs hook that generates documentation for each specification in
     # +specs+
  
  -  def self.generate installer, specs
  +  def self.generation_hook installer, specs
       start = Time.now
       types = installer.document
  
  @@ -77,10 +64,6 @@ class RDoc::RubyGemsHook
       say "Done installing documentation for #{names} after #{duration} seconds"
     end
  
  -  def self.remove uninstaller
  -    new(uninstaller.spec).remove
  -  end
  -
     ##
     # Loads the RDoc generator
  
  @@ -263,15 +246,3 @@ class RDoc::RubyGemsHook
  ```

</details>

Reply:
This is an outdated comment now.
Because we reverted the moving `RDoc::RubyGemsHook` to `lib/rubygems_plug.rb` from `lib/rdoc/rubygems_hook.rb` to support installed as a default gem.

Reply:
Thank you for your additional explanation.